packages/smithy-client/src/split-header.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @param value - header string value.
+ * @returns value split by commas that aren't in quotes.
+ */
+export const splitHeader = (value: string): string[] => {
+  const z = value.length;
+  const values = [];
+
+  let withinQuotes = false;
+  let prevChar = undefined;
+  let anchor = 0;
+
+  for (let i = 0; i < z; ++i) {
+    const char = value[i];
+    switch (char) {
+      case `"`:
+        if (prevChar !== "\\") {
+          withinQuotes = !withinQuotes;
+        }
+        break;
+      case ",":
+        if (!withinQuotes) {
+          values.push(value.slice(anchor, i));
+          anchor = i + 1;
+        }
+        break;
+      default:
+    }
+    prevChar = char;
+  }
+
+  values.push(value.slice(anchor));
+
+  return values.map((v) => {
+    v = v.trim();
+    const z = v.length;
+    if (v[0] === `"` && v[z - 1] === `"`) {
```

Comment:
Can `v` be empty? What if there's sequential `,`, or `,` only separated by ws?

Reply:
added more test cases to describe what happens, i.e. `,,` is 3 empty strings, and so is `  ,   ,   `.